### PR TITLE
chore(internal): consolidate git utils

### PR DESCRIPTION
**Pull request by namjul**
The aim of this PR is to consolidate spread out common git utils `common-all`.
git utils inside `common-all` are re-exported in git utils `common-server` thereby nothing changes for the rest of the code. Its also practicall to have `common-all` git utils included in `common-server`.

# Pull Request Checklist

If you are a community contributor, copy and paste the PR template from [Dendron Community PR Checklist](https://gist.github.com/kevinslin/5d1a6663638259e7dbd33b01975cc00f) and add it to the body of the pull request. 

If you are a team member, copy and paste the template from [Dendron Extended PR Checklist](https://gist.github.com/kevinslin/dfc7a101f21c58478216aa0d70256bc2).

To copy the template, click on the `Raw` button on the gist to copy the plaintext version of the template to this PR.

### CODE REVIEW

**Review by hikchoi — COMMENTED** [review]
looks like one test related to git is failing.

**Comment by kevinslin** [comment]
looks like test still failing

**Comment by namjul** [comment]
The issue is that the following code does not really stub `getGitProviderOwnerAndRepository`
https://github.com/dendronhq/dendron/blob/35e0e6730129d246e2ff515fb6c8b190abf7b157/packages/plugin-core/src/test/suite-integ/CopyCodespaceURL.test.ts#L21-L23

Looking further into this.

File: packages/common-all/src/git.ts [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+/**
+ *  NOTICE: Lots of the Git code is obtained from https://github.com/KnisterPeter/vscode-github, licened under MIT
+ */
+
+/**
+ * Utilities for working with git urls
+ */
+
+export function canShowGitLink(opts: {
```
Comment (kevinslin) [inline packages/common-all/src/git.ts:23]: can we add some tests for these methods?

**Review by kevinslin — CHANGES_REQUESTED** [review]

File: packages/common-all/src/git.ts (hunk 1 above)
Comment (namjul) [reply packages/common-all/src/git.ts:23]: Bootstaped with `canShowGitLink` util https://github.com/dendronhq/dendron/pull/3681/commits/bfb2d56d5c982cbb2409839f04eda965d03fc94b

Shall I continue adding tests for the rest of the utils?

**Review by kevinslin — APPROVED** [review]

File: packages/common-all/src/git.ts (hunk 1 above)
Comment (namjul) [reply packages/common-all/src/git.ts:23]: Added a couple more but leave it now as it is [`0e6b42a` (#3681)](https://github.com/dendronhq/dendron/pull/3681/commits/0e6b42ab7f8297e0a46fd3421e1dbc37b6b3e739)

**Pull request merged**